models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+  # place. The basic flow will be like this:
+  #  - Read dvmdostem parameter file into json object, load into memory
+  #  - Update the in-memory json object
+  #  - Write the json object back out to a new dvmdostem parameter file
+
+  # Start by figuring out which community number we are working on.
+  # This is important because we need to pull the correct "community block"
+  # from the dvmdostem parameter files. Each trait in the traits dataframe
+  # should have a name like this:
+  #   CMT04.Salix.extinction_coefficient_diffuse
+  # So we can parse the first name in the dataframe to determine the CMT number.
+  # !! Actually, it should just be CMT04.Salix OR extinction_coefficient_diffuse
+  # !! Can't have PFT name linked with trait otherwise that would require a lot of 
+  # !! hard coding in this function. 
+  cmtnum <- strsplit(unlist(strsplit(names(trait_df)[1], '.', fixed=TRUE))[1], "CMT")
+  cmtnum <- unlist(cmtnum)[2]
+  cmtnum <- as.numeric(cmtnum)
+
+  # Next, use a helper script distributed with dvmdostem to read the dvmdostem
+  # parameter data into memory as a json object, using a temporaroy json file
+  # to hold a representation of each dvmdostem parameter file.
+  dimveg_params <- paste(appbinary_path, "parameters", 'cmt_dimvegetation.txt', sep="/")
+  envcanopy_params <- paste(appbinary_path, "parameters", 'cmt_envcanopy.txt', sep="/")
+
+  dimveg_jsonfile <- '/tmp/dvmdostem-dimveg.json'
+  envcanopy_jsonfile <- '/tmp/dvmdostem-envcanopy.json'
+
+  # Call the helper script and write out the data to a temporary file
+  # This gets just the block we are interested in (based on community type)
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--dump-block-to-json", dimveg_params, cmtnum)),
+          stdout=dimveg_jsonfile, wait=TRUE)
+
+  system2(paste0(appbinary_path,"/scripts/param_util.py"),
+          args=(c("--dump-block-to-json", envcanopy_params, cmtnum)),
+          stdout=envcanopy_jsonfile, wait=TRUE)
+
+  # Read the json file into memory
+  library("rjson")
+  dimveg_jsondata <- fromJSON(paste(readLines(dimveg_jsonfile), collapse=""))
+  envcanopy_jsondata <- fromJSON(paste(readLines(envcanopy_jsonfile), collapse=""))
+
+  # (2)
+  # Overwrite certain parameter values with (ma-posterior) trait data
+  # from pecan, then write back out to disk...
+
+  # loop over all the incoming samples (traits)
+  for (cur_t_name in names(trait_df)) {
+    #print(cur_t_name)
+
+    # In converting trait.values into a dataframe, we get 
+    # these long column names that are dot separted and seem
+    # to contain cmtkey.pftname.traitname
+    # So the following splits out the various components into 
+    # convenient local variables
+    cmtkey <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[1]
+    pftname <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[2]
+    tname <- unlist(strsplit(cur_t_name, '.', fixed=TRUE))[3]
```

Comment:
If you do need to do this (which still seems wrong to me), I'd at least do the ```unlist(strsplit(cur_t_name, '.', fixed=TRUE))``` once to a variable, and then use that variable to extract the 3 parts you need.